static/src/javascripts/projects/common/modules/crosswords/grid.js

```diff
@@ -0,0 +1,176 @@
+        const borderWidth = 1;
+
+        if (direction === 'across') {
+            const width = 1;
+            return React.createElement('rect', {
+                x: left - borderWidth - width,
+                y: top,
+                key: ['sep', direction, x, y].join('_'),
+                width,
+                height: constants.cellSize,
+            });
+        } else if (direction === 'down') {
+            const height = 1;
+            return React.createElement('rect', {
+                x: left,
+                y: top - borderWidth - height,
+                key: ['sep', direction, x, y].join('_'),
+                width: constants.cellSize,
+                height,
+            });
+        }
+    }
+
+    // Position in-between this and previous cells
+    static createHyphenSeparator(
+        x: number,
+        y: number,
+        direction: string
+    ): ?Element {
+        const top = helpers.gridSize(y);
+        const left = helpers.gridSize(x);
+        const borderWidth = 1;
+        let width;
+        let height;
+
+        if (direction === 'across') {
+            width = constants.cellSize / 4;
+            height = 1;
+            return React.createElement('rect', {
+                x: left - borderWidth / 2 - width / 2,
+                y: top + constants.cellSize / 2 + height / 2,
+                width,
+                height,
+            });
+        } else if (direction === 'down') {
+            width = 1;
+            height = constants.cellSize / 4;
+            return React.createElement('rect', {
+                x: left + constants.cellSize / 2 + width / 2,
+                y: top - borderWidth / 2 - height / 2,
+                width,
+                height,
+            });
+        }
+    }
+
+    static createSeparator(
+        x: number,
+        y: number,
+        separatorDescription: ?Object
+    ): ?Element {
+        if (separatorDescription) {
+            if (separatorDescription.separator === ',') {
+                return Grid.createWordSeparator(
+                    x,
+                    y,
+                    separatorDescription.direction
+                );
+            } else if (separatorDescription.separator === '-') {
+                return Grid.createHyphenSeparator(
+                    x,
+                    y,
+                    separatorDescription.direction
+                );
+            }
+        }
+    }
+
+    getSeparators(x: number, y: number): ?Object {
+        return this.props.separators[helpers.clueMapKey(x, y)];
+    }
+
+    handleSelect(x: number, y: number): void {
+        this.props.crossword.onSelect(x, y);
+    }
+
+    render(): Element {
+        const width = helpers.gridSize(this.props.columns);
+        const height = helpers.gridSize(this.props.rows);
+        const cells = [];
+        let separators = [];
+
+        const range = n => Array.from({ length: n }, (value, key) => key);
+
+        const self = this;
```

Comment:
I don't think we need to do this, as arrow functions propagate the context through the call stack

Reply:
Ah, yea, sorry, that was left from when I was trying to deal with something else. Will remove.